File: src/com/anathema_roguelike/entities/characters/perks/actions/TargetedPerk.java
package com.anathema_roguelike.entities.characters.perks.actions;

import com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.Targetable;
import com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.TargetingStrategy;
import com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.ranges.Range;

public abstract class TargetedPerk<T extends Targetable> extends GenericTargetedPerk<T, T> {

	public TargetedPerk(String name, Range<T> range, TargetingStrategy<T, T> strategy) {
		super(name, range, strategy);
	}
}

File: src/com/anathema_roguelike/entities/characters/perks/actions/TargetedPerk.scala


package com.anathema_roguelike
package entities.characters.perks.actions

import com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.{TargetFilter, Targetable}
import com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.ranges.Range
import com.anathema_roguelike.entities.characters.perks.requirements.PerkRequirement

abstract class TargetedPerk[T <: Targetable](
  name: String,
  range: Range[T],
  strategy: TargetFilter[T, T],
  requirements: PerkRequirement*)

  extends GenericTargetedPerk[T, T](name, range, strategy, requirements:_*) {

}
